packages/react-events/src/Press.js

```diff
@@ -0,0 +1,332 @@
+  isPressed: boolean,
+  longPressTimeout: null | TimeoutID,
+  pressTarget: null | EventTarget,
+};
+
+function dispatchPressEvent(
+  context: EventResponderContext,
+  name: string,
+  state: PressState,
+  listener: (e: Object) => void,
+): void {
+  context.dispatchEvent(name, listener, state.pressTarget, true);
+}
+
+function dispatchPressInEvents(
+  context: EventResponderContext,
+  props: Object,
+  state: PressState,
+): void {
+  if (props.onPressIn) {
+    context.dispatchEvent('pressin', props.onPressIn, state.pressTarget, true);
+  }
+  if (props.onPressChange) {
+    const pressChangeEventListener = () => {
+      props.onPressChange(true);
+    };
+    context.dispatchEvent(
+      'presschange',
+      pressChangeEventListener,
+      state.pressTarget,
+      true,
+    );
+  }
+  if (!state.isLongPressed && (props.onLongPress || props.onLongPressChange)) {
+    const longPressDelay = props.longPressDelay || 1000;
+    state.longPressTimeout = setTimeout(() => {
+      state.isLongPressed = true;
+      state.longPressTimeout = null;
+      if (props.onLongPressChange) {
+        const longPressChangeEventListener = () => {
+          props.onLongPressChange(true);
+        };
+        context.dispatchEvent(
+          'longpresschange',
+          longPressChangeEventListener,
+          state.pressTarget,
+          true,
+        );
+      }
+    }, longPressDelay);
+  }
+}
+
+function dispatchPressOutEvents(
+  context: EventResponderContext,
+  props: Object,
+  state: PressState,
+): void {
+  if (state.longPressTimeout !== null) {
+    clearTimeout(state.longPressTimeout);
+    state.longPressTimeout = null;
+  }
+  if (props.onPressOut) {
+    context.dispatchEvent(
+      'pressout',
+      props.onPressOut,
+      state.pressTarget,
+      true,
+    );
+  }
+  if (props.onPressChange) {
+    const pressChangeEventListener = () => {
+      props.onPressChange(false);
+    };
+    context.dispatchEvent(
+      'presschange',
+      pressChangeEventListener,
+      state.pressTarget,
+      true,
+    );
+  }
+  if (props.onLongPressChange && state.isLongPressed) {
+    const longPressChangeEventListener = () => {
+      props.onLongPressChange(false);
+    };
+    context.dispatchEvent(
+      'longpresschange',
+      longPressChangeEventListener,
+      state.pressTarget,
+      true,
+    );
+  }
+}
+
+function isAnchorTagElement(eventTarget: EventTarget): boolean {
+  return (eventTarget: any).nodeName === 'A';
+}
+
+const PressResponder = {
+  targetEventTypes,
+  createInitialState(): PressState {
+    return {
+      defaultPrevented: false,
+      isAnchorTouched: false,
+      isLongPressed: false,
+      isPressed: false,
+      longPressTimeout: null,
+      pressTarget: null,
+    };
+  },
+  handleEvent(
+    context: EventResponderContext,
+    props: Object,
+    state: PressState,
+  ): void {
+    const {eventTarget, eventType, event} = context;
+
+    switch (eventType) {
+      case 'keydown': {
```

Comment:
I think we should consider making keyboards trigger `pressin`, `pressout`, `press`, and even `longpress` events.

Reply:
Will do this at a later point, in a follow up.